File: 217_Contains Duplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function(nums) {
    let map = new Map();
    const size = nums.length;
    
    for(let i = 0; i < size; ++i) {
        if(!map.get(nums[i])) map.set(nums[i], 1);
        else return true;
    }
    return false;
};

File: 217_Contains Duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        std::unordered_set<int> set;
        
        for(size_t i = 0; i < nums.size(); ++i) {
            if(set.find(nums[i]) != set.end()) {
                return true;
            } else {
                set.insert(nums[i]);
            }
        }
        return false;
    }
};